paddle/phi/kernels/onednn/transpose_kernel.cc

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/phi/kernels/transpose_kernel.h"
+#include "paddle/fluid/framework/tensor_util.h"
```

Comment:
If you use copy function in PHI, please use `copy` in "paddle/phi/core/tensor_utils.h"

Reply:
@YuanRisheng There is no Copy equivalent for OneDNNContext, should I leave it like this or create OneDNNContext version in paddle/phi/core/tensor_utils.cc? I tried to do it in this [PR](https://github.com/PaddlePaddle/Paddle/pull/46139/commits/ebd80bf7339e3082478bb99aa1f1648544d1cd46) but then Copy behaved inconsistently with OneDnn

Reply:
Oh, I see. The code of MKLDNN Copy is not implemented in paddle/phi/core/tensor_utils.cc. You may need migrate the code of MKLDNN Copy(The code is in TensorCopyImpl function) from paddle/fluid/framework/tensor_util.cc to paddle/phi/core/tensor_utils.cc. This is a necessary work for MKLDNN kernel migration. Thank you.

Reply:
@YuanRisheng I think the problem with phi::Copy may have occured in previous PRs (e.g. transpose2_grad) and there TensorCopy was used instead of phi::Copy, so if I add OneDNNContext version I shoul also change it in other kernels. Can I leave TensorCopy in this PR and dedicate seperate PR for creating OneDNN version of phi::Copy?

Reply:
That's Ok.

Reply:
@YuanRisheng so could you give me approve as the check failed because that TensorCopy usage? Could you also take a look at CE-FRAMEWORK please? It fails even though there are no errors and all tests pass. Besides these two checks I think this PR is ready to merge. Thanks!

Reply:
The CE-FRAMEWORK seems have a error. It is a OneDNN error that is unfamiliar to me. I am not sure whether this error is related to the file "operator_reshape2_onednn_fuse_pass.cc" you modified.  Maybe you can rerun the CE-FRAMEWORK second time. If CE-FRAMEWORK fails again, you need deal with this issue. After all CI pass(except CI-APPROVAL), I will approve and merge this PR. Thanks.
<img width="897" alt="image" src="https://user-images.githubusercontent.com/29249150/201908860-ed1237d4-bfbf-4e00-8f18-2d9cc97a82af.png">

Reply:
Hi @YuanRisheng could you somehow share with me this script which is failing (`test_swin_transformer`)? We do not have it locally and in this case it is really hard to guess what went wrong.

Reply:
@YuanRisheng now tests pass except coverage, because kernel is probably not fully covered by unit tests in fluid. May I get an approve for TensorCopy please?

Reply:
That's OK and please create OneDNN version of phi::Copy in next PR. Thank you